Lib/fontTools/cffLib/transforms.py

```diff
@@ -362,8 +362,7 @@ def remove_hints(cff):
         #    we have seen any non-hint operators so far and do the right
         #    thing, recursively... Good luck understanding that :(
         css = set()
-        for g in font.charset:
-            c, _ = cs.getItemAndSelector(g)
+        for c in cs.values():
```

Comment:
Might warrant a test, if possible, since this was not caught earlier.

Reply:
Will add. Is probably enough to run on a CFF2.